src/config_file.c

```diff
@@ -121,21 +117,21 @@ static void config_file_clear(diskfile *file)
 
 static int config_open(git_config_backend *cfg, git_config_level_t level, const git_repository *repo)
 {
-	diskfile_backend *b = GIT_CONTAINER_OF(cfg, diskfile_backend, header.parent);
+	diskfile_backend *b = GIT_CONTAINER_OF(cfg, diskfile_backend, parent);
 	int res;
 
-	b->header.level = level;
-	b->header.repo = repo;
+	b->level = level;
+	b->repo = repo;
 
-	if ((res = git_config_entries_new(&b->header.entries)) < 0)
+	if ((res = git_config_entries_new(&b->entries)) < 0)
 		return res;
 
 	if (!git_path_exists(b->file.path))
 		return 0;
 
-	if (res < 0 || (res = config_read(b->header.entries, repo, &b->file, level, 0)) < 0) {
-		git_config_entries_free(b->header.entries);
-		b->header.entries = NULL;
+	if (res < 0 || (res = config_read(b->entries, repo, &b->file, level, 0)) < 0) {
```

Comment:
I don't think `res` can be negative here, as l126 would have handled it.

Reply:
True. I've tried to refrain from updating existing code though to more clearly highlight what changed across the moves.